File: ruby/035.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer}
def search_insert(nums, target)
  return 0 if target <= nums[0]
  return nums.length if target > nums[-1]

  i = 1
  while nums[i] < target do
    i += 1
  end

  i
end

File: javascript/035.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function(nums, target) {
  if (target <= nums[0]) return 0;
  if (target > nums[nums.length - 1]) return nums.length;
  
  var i = 1;
  while(target > nums[i]) { ++i; }
  
  return i;
};